crates/gossip/src/gossip.rs

```diff
@@ -91,29 +93,32 @@ impl Drop for DropAbortHandle {
 ///
 /// This type acts as both an implementation of the [Gossip] trait and a [TxModuleHandler].
 #[derive(Debug, Clone)]
-struct K2Gossip {
+pub(crate) struct K2Gossip {
     config: Arc<K2GossipConfig>,
     /// The state of the current initiated gossip round.
     ///
     /// We only initiate one round at a time, so this is a single value.
-    initiated_round_state: Arc<Mutex<Option<GossipRoundState>>>,
+    pub(crate) initiated_round_state: Arc<Mutex<Option<GossipRoundState>>>,
     /// The state of currently accepted gossip rounds.
     ///
     /// This is a map of agent ids to their round state. We can accept gossip from multiple agents
     /// at once, mostly to avoid coordinating initiation.
-    accepted_round_states: Arc<Mutex<HashMap<Url, GossipRoundState>>>,
+    pub(crate) accepted_round_states:
+        Arc<Mutex<HashMap<Url, GossipRoundState>>>,
     space_id: SpaceId,
     // This is a weak reference because we need to call the space, but we do not create and own it.
     // Only a problem in this case because we register the gossip module with the transport and
     // create a cycle.
-    peer_store: DynPeerStore,
-    local_agent_store: DynLocalAgentStore,
-    peer_meta_store: Arc<K2PeerMetaStore>,
+    pub(crate) peer_store: DynPeerStore,
+    pub(crate) local_agent_store: DynLocalAgentStore,
+    pub(crate) peer_meta_store: Arc<K2PeerMetaStore>,
     op_store: DynOpStore,
     fetch: DynFetch,
     agent_verifier: DynVerifier,
     response_tx: Sender<GossipResponse>,
     _response_task: Arc<DropAbortHandle>,
+    _initiate_task: Arc<Option<DropAbortHandle>>,
+    _timeout_task: Arc<Option<DropAbortHandle>>,
```

Comment:
This task spawning business leads to strange fields, same in the fetch module.

Reply:
It does, yes. They're really just there to link the life cycles together but aren't useful in their own right... If I think of a nicer way to express the same thing I will go for it, but I have no ideas at the moment :)